### ISSUE TITLE
Running App in Debug Mode from Visual Studio Installs in Solution Directory

### ISSUE THREAD

**Issue by veevandyke** [Squirrel/Squirrel.Windows#1612]
**Squirrel version(s)**
1.9.1
**Description**
Running my app in debug mode silently installs the app, but in my solution directory.

**Steps to recreate**
I'm not sure exactly how this happened, but I noticed that there was an extra installation in the control panel. I uninstalled both of the installations, and it deleted everything in my solution directory. All code pages, solution files, even my .git folder. I recreated my project using a decompiler and my .git remote backup. (Thankfully only a few days old as the local .git folder to which I had been committing was now gone,)

I had no installations of my app on my dev computer. After running the app in debug mode from Visual Studio (2019), I noticed an additional app-8.0.0 directory in my solution folder and sure enough, there is an instance of the app in the control panel again. (Which I will NOT be uninstalling to ensure that my files don't disappear again.)

I am not sure if there is something in the updatemanager code that is doing this, but what I really don't understand is WHY it's putting it in my solution directory as opposed to my localappdata the way it's supposed to.

**Expected behavior**
Should not install when running in debug mode. DEFINITELY should not install to code solution directory. DEFINITELY should not delete code files when uninstalling.

**Actual behavior**
See Expected Behavior above.

See solution directory with app-8.0.0 directory. NOTE: Solution directory is NOT in localappdata. It's in a mapped network drive (F)

![Squirrel-Installation](https://user-images.githubusercontent.com/7154028/78807942-c1184200-7992-11ea-9888-2dde8bc2c049.png)

**Comment by anaisbetts** [comment]
So sorry that you ended up deleting your code this way! I'm not sure what's going on either, but it's best to gate the Squirrel code to detect whether it's installed or not (I believe there is a method to tell you whether you're an installed app or not). Unfortunately, it's not possible for a library to detect whether its caller is in Debug Mode or not.

**Comment by Horb** [comment]
What does your squirrel integration code look like?

**Comment by veevandyke** [comment]
Thanks @anaisbetts. I will definitely do that, but I'm still not sure WHY it installed from there in the first place. I do actually have code in my Squirrel hooks to kill the procedure if I am in debug mode. 

Interestingly, there is nothing in the SquirrelTemp log to indicate the installation in my solution directory either. 

I actually copied my code to a backup folder and uninstalled it again (yes, it wiped out my code again.) and despite no changes to the Squirrel code, it has NOT installed itself again. Not sure why it did it twice, but it's not happening now.

@Horb My basic integration code is thus:

```
`        Private Sub MyApplication_Startup(sender As Object, e As StartupEventArgs) Handles Me.Startup
            Try
                SquirrelAwareApp.HandleEvents(onInitialInstall:=AddressOf squirrelInitialInstall, onAppUpdate:=AddressOf squirrelAppUpdate, onAppUninstall:=AddressOf squirrelAppUninstall, onFirstRun:=AddressOf squirrelFirstRun)
            Catch ex As Exception
                Application.Log.WriteException(ex)
            End Try
            Application.EnableVisualStyles() = True
        End Sub`
```

Nothing fancy. The actual code does some folder copying, registry key creating, etc. and creates shortcuts in all various locations. (Sorry. It's inherited. I looked at converting to C# a while back, but there is just WAY too much code so I'm stuck with VB.net)

Let me know if there is something specific you want to see to see if it gives you a clue as to why this happened. :) More than anything, I just don't want this to happen to anyone else. (There was a significant amount of cursing and head-pounding-on-desk when I realized that my code files were gone.)
